### PR TITLE
fix: several lucky drop issues

### PR DESCRIPTION
**Pull request by hanszhoou**
## Description

Closes # (NO_ISSUE)
https://mask.atlassian.net/browse/MASK-762
https://mask.atlassian.net/browse/MASK-816
https://mask.atlassian.net/browse/MASK-418
https://mask.atlassian.net/browse/MASK-836

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

**Comment by wenluomask** [comment]
Checked.

File: packages/mask/src/plugins/RedPacket/SNSAdaptor/SelectNftTokenDialog.tsx [hunk 1]
```diff
@@ -758,11 +754,10 @@ function NFTCard(props: NFTCardProps) {
                     loadingFailImage: classes.loadingFailImage,
                     iframe: classes.iframe,
                 }}
-                setERC721TokenName={setName}
             />
             <div className={classes.selectWrapperNftNameWrapper}>
                 <Typography className={classes.selectWrapperNftName} color="textSecondary">
-                    {name}
+                    {'#' + token.tokenId}
```
Comment (guanbinrui) [inline packages/mask/src/plugins/RedPacket/SNSAdaptor/SelectNftTokenDialog.tsx]: We should not suppose that any NFT token uses incremental numbers as token IDs by default. Instead, some NFT tokens use hash strings.

Comment (hanszhoou) [reply packages/mask/src/plugins/RedPacket/SNSAdaptor/SelectNftTokenDialog.tsx]: As I know that hash string is hex number. Do you think different NFT with the same name is ok?

Comment (guanbinrui) [reply packages/mask/src/plugins/RedPacket/SNSAdaptor/SelectNftTokenDialog.tsx]: That could be a long long string. Like this one: https://opensea.io/assets/0x699aa5e6eab40bcda7398ec7e5d585ca5d656a13/60212408056462910084986498866292153316932473047591013797793482224010201900028

Comment (guanbinrui) [reply packages/mask/src/plugins/RedPacket/SNSAdaptor/SelectNftTokenDialog.tsx]: It needs to be formated just like addresses.

Comment (hanszhoou) [reply packages/mask/src/plugins/RedPacket/SNSAdaptor/SelectNftTokenDialog.tsx]: Yeah, I will discuss this to PM.

**Comment by wenluomask** [comment]
Checked.

**Review by Jack-Works — DISMISSED** [review]

File: packages/mask/src/plugins/RedPacket/SNSAdaptor/SelectNftTokenDialog.tsx (hunk 1 above)
Comment (hanszhoou) [reply packages/mask/src/plugins/RedPacket/SNSAdaptor/SelectNftTokenDialog.tsx]: image

File: packages/mask/src/plugins/RedPacket/SNSAdaptor/SelectNftTokenDialog.tsx [hunk 2]
```diff
@@ -758,11 +755,10 @@ function NFTCard(props: NFTCardProps) {
                     loadingFailImage: classes.loadingFailImage,
                     iframe: classes.iframe,
                 }}
-                setERC721TokenName={setName}
             />
             <div className={classes.selectWrapperNftNameWrapper}>
                 <Typography className={classes.selectWrapperNftName} color="textSecondary">
-                    {name}
+                    {'#' + formatNFT_TokenId(token.tokenId, 2)}
```
Comment (guanbinrui) [inline packages/mask/src/plugins/RedPacket/SNSAdaptor/SelectNftTokenDialog.tsx]: Why not add the `#` prefix in `formatNFT_TokenId`? Additionally, could you please refactor other places by using this way?

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**